### PR TITLE
[otci] add link metrics commands to otci

### PR DESCRIPTION
**Pull request by zhanglongxia**

### CODE REVIEW

File: tools/otci/otci/otci.py [hunk 1]
```diff
@@ -2148,13 +2148,150 @@ def set_domain_name(self, name: str):
...
+    def __valid_flags(self, flags: str, flags_set: str):
+        # check for duplicate chars
+        if len(flags) != len(set(flags)):
+            return False
+
+        for c in flags:
+            if c not in flags_set:
+                return False
+
+        return True
```
Comment (superwhd) [inline tools/otci/otci/otci.py:2294]: what about this:
```suggestion
   return set(flags).issubset(set(flags_set))
```

File: tools/otci/otci/otci.py [hunk 2]
```diff
@@ -2148,13 +2148,150 @@ def set_domain_name(self, name: str):
...
+            elif '- RSSI' in line:
+                results['rssi'] = self.__parse_line_numbers(line)[0]
+            elif '- PDU Counter' in line:
+                results['pdu_counter'] = self.__parse_line_numbers(line)[0]
+            else:
+                continue
+
+        return results
+
+    def __parse_line_numbers(self, line: str) -> List[int]:
```
Comment (superwhd) [inline tools/otci/otci/otci.py:2276]: The method's name makes me think it's parsing the line number.

File: tools/otci/otci/otci.py [hunk 3]
```diff
@@ -2148,13 +2148,150 @@ def set_domain_name(self, name: str):
...
+        results = []
+        values = re.findall("\-?\d+", line)
+
+        for value in values:
+            results.append(int(value))
+
+        return results
```
Comment (superwhd) [inline tools/otci/otci/otci.py:2283]:
```suggestion
        return list(map(int, results))
```

File: tools/otci/otci/otci.py [hunk 4]
```diff
@@ -2148,13 +2148,150 @@ def set_domain_name(self, name: str):
...
+            if 'Received Link Metrics Management Response from' in line:
+                address = line.split(': ')[1].strip()
+                report_received = address == peer_addr
+            elif 'Status' in line:
+                status = line.split(':')[1].strip()
+
+        if report_received and status == 'Success':
+            ret = True
+
+        return ret
```
Comment (superwhd) [inline tools/otci/otci/otci.py:2219]:
```suggestion
        return report_received and status == 'Success'
```

**Review by superwhd — APPROVED** [review]
LGTM overall with some minor suggestions

File: tools/otci/otci/otci.py (hunk 2 above)
Comment (zhanglongxia) [reply tools/otci/otci/otci.py]: I have updated the name to `__parse_numbers`

File: tools/otci/otci/otci.py [hunk 5]
```diff
@@ -2148,13 +2148,138 @@ def set_domain_name(self, name: str):
...
+            if '- LQI' in line:
+                results['lqi'] = self.__parse_numbers(line)[0]
+            elif '- Margin' in line:
+                results['margin'] = self.__parse_numbers(line)[0]
+            elif '- RSSI' in line:
+                results['rssi'] = self.__parse_numbers(line)[0]
+            elif '- PDU Counter' in line:
+                results['pdu_counter'] = self.__parse_numbers(line)[0]
+            else:
+                continue
```
Comment (Irving-cl) [inline tools/otci/otci/otci.py:2269]: Is this required?

**Review by Irving-cl — APPROVED** [review]
LGTM

**Review by abtink — APPROVED** [review]
👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**